test/org/centos/contra/Infra/Utils.groovy

```diff
@@ -0,0 +1,347 @@
+import spock.lang.*
+import com.homeaway.devtools.jenkins.testing.JenkinsPipelineSpecification
+
+import org.yaml.snakeyaml.Yaml
+
+public class UtilsSpec extends JenkinsPipelineSpecification {
+    @Shared
+    def config = new HashMap<String, ?>()
+
+    @Shared
+    def infraUtils
+
+    def setup() {
+        infraUtils = loadPipelineScriptForTest("../../src/org/centos/contra/Infra/Utils.groovy")
+        explicitlyMockPipelineStep('dir')
+        explicitlyMockPipelineStep('writeFile')
+        explicitlyMockPipelineStep('fileExists')
+        explicitlyMockPipelineStep('withEnv')
+        explicitlyMockPipelineStep('container')
+
+        explicitlyMockPipelineVariable('WORKSPACE')
```

Comment:
In my experience, you don't need to explicitly mock variables that you expect to be available from jenkins. You can do `infraUtils.getBinding().setVariable("WORKSPACE", "workspace")`, ect.